Explorer/Assets/DCL/UI/MainUIContainer/MainUIController.cs

```diff
@@ -0,0 +1,156 @@
+            viewInstance.pointerDetectionArea.OnEnterArea += OnPointerEnter;
+            viewInstance.pointerDetectionArea.OnExitArea += OnPointerExit;
+            mvcManager.ShowAsync(SidebarController.IssueCommand()).Forget();
+            mvcManager.ShowAsync(MinimapController.IssueCommand()).Forget();
+            mvcManager.ShowAsync(ChatController.IssueCommand()).Forget();
+            showingSidebar = true;
+        }
+
+        private void OnSidebarAutohideStatusChanged(bool status)
+        {
+            autoHideSidebar = status;
+            hideSidebarCancellationTokenSource = hideSidebarCancellationTokenSource.SafeRestart();
+            showSidebarCancellationTokenSource = showSidebarCancellationTokenSource.SafeRestart();
+        }
+
+
+        private void OnSidebarBlockStatusChanged(bool status)
+        {
+            sidebarBlockStatus = status;
+            hideSidebarCancellationTokenSource = hideSidebarCancellationTokenSource.SafeRestart();
+            showSidebarCancellationTokenSource = showSidebarCancellationTokenSource.SafeRestart();
+        }
+
+        private void OnPointerEnter()
+        {
+            if (autoHideSidebar || sidebarBlockStatus) return;
+
+            if (showingSidebar) { waitingToShowSidebar = false; }
+
+            if (showSidebarCancellationTokenSource.IsCancellationRequested) { waitingToShowSidebar = false; }
+
+            if (!showingSidebar && !waitingToShowSidebar)
+            {
+                waitingToShowSidebar = true;
+                showSidebarCancellationTokenSource = showSidebarCancellationTokenSource.SafeRestart();
+                WaitAndShow(showSidebarCancellationTokenSource.Token).Forget();
+            }
+
+            if (waitingToHideSidebar) { hideSidebarCancellationTokenSource.Cancel(); }
+        }
+
+        private void OnPointerExit()
+        {
+            if (autoHideSidebar || sidebarBlockStatus) return;
+
+            if (waitingToShowSidebar || showingSidebar) { showSidebarCancellationTokenSource.Cancel(); }
+
+            if (hideSidebarCancellationTokenSource.IsCancellationRequested) { waitingToHideSidebar = false; }
+
+            if (!waitingToHideSidebar && showingSidebar)
+            {
+                waitingToHideSidebar = true;
+                hideSidebarCancellationTokenSource = hideSidebarCancellationTokenSource.SafeRestart();
+                WaitAndHide(hideSidebarCancellationTokenSource.Token).Forget();
+            }
+        }
+
+        private async UniTask WaitAndHide(CancellationToken ct)
+        {
+            await UniTask.Delay(TimeSpan.FromSeconds(HIDE_SIDEBAR_WAIT_TIME), cancellationToken: ct);
+            waitingToHideSidebar = false;
+            if (ct.IsCancellationRequested) return;
+
+            await AnimateWidthAsync(HIDE_SIDEBAR_LAYOUT_WIDTH, ct);
+
+            if (ct.IsCancellationRequested) return;
+            showingSidebar = false;
+        }
+
+        private async UniTask WaitAndShow(CancellationToken ct)
```

Comment:
Perhaps you can change it to `UniTaskVoid` which is more performant when doing `Forget()`